Monika After Story/game/definitions.rpy

```diff
@@ -1287,6 +1287,10 @@ default persistent.gender = "M" #Assume gender matches the PC
 default persistent.chess_strength = 3
 default persistent.closed_self = False
 default persistent.seen_monika_in_room = False
+default persistent.ever_won = {'pong':False,'chess':False}
+default persistent.sessions=[]
```

Comment:
Tracking session data could give us some interesting potential (like graphing, or something statistical), but I'm concerned about _the growing size of persistent_. Maybe consider a cap on the amount of saved sessions?

Reply:
What would be a reasonable cap?

Reply:
`time.time()` objects are 24 bytes. An empty list is 72 bytes, and with 2 items in it, it increases to 88 bytes. That's 136 bytes per session. If we assume that the user opens monika twice a day (once in the morning, once at night), thats 272 bytes per day. In a month that results in 8KB. It doesn't sound like a lot, but it's not fixed, which allows persistent (as well as RAM) usage to grow. Generally, we should avoid having items that could grow without a safe guard in case it gets too large. 

In this case we should consider what is the end goal of collecting this data. If we wanted an average time spent with monika, we could simply keep track of the total time as well as the number of sessions. If we wanted average start time, we could we keep track of the average starttime as well as the number of sessions. (same with end time) These cases would leave us with a constantly growing integer/double value, but those have fixed sizes, so it isn't a memory problem anymore.

If keeping track of the specific start and end times is a necessity, then I suggest we cap the list at 10,000 elements. This translates to 1.36 MB of data, which quite large for a simple game but not intrusive. 

In addition, you should change the storage of each start/end time pair from a list into a `tuple`. (so `persistent.sessions` stores a list of tuples instead of a list of lists). Tuples have a smaller memory footprint because they have fixed sizes.